photon-lib/src/main/scala/com/linkedin/photon/ml/function/L2Regularization.scala

```diff
@@ -26,6 +27,7 @@ import com.linkedin.photon.ml.util.BroadcastWrapper
 trait L2Regularization extends ObjectiveFunction {
 
   protected var l2RegWeight: Double = 0D
+  protected var interceptIndex: Int = -1
```

Comment:
I think that we should have this be an `Option` instead. It will simplify the safety check and save us from having to declare a constant with value `-1` which is then referenced by `ObjectiveFunctionHelper` and the loss function classes.

In this case, we should also rename the member to `interceptIndexOpt`.